pkg/bmc/idrac_virtualmedia.go

```diff
@@ -90,3 +91,12 @@ func (a *redfishiDracVirtualMediaAccessDetails) VendorInterface() string {
 func (a *redfishiDracVirtualMediaAccessDetails) SupportsSecureBoot() bool {
 	return true
 }
+
+func (a *redfishiDracVirtualMediaAccessDetails) Validate(bmcCreds Credentials) error {
+	address := getRedfishAddress(a.bmcType, a.host)
+	vmediahttpClient := http.Client{
```

Comment:
I'm not sure it could be a good strategy to create a new client instance every time in the `Validate` method, rather than injecting from outside. For example, the Ironic provisioner already creates a singleton client for caching and reusing connections (see https://github.com/metal3-io/baremetal-operator/blob/a86fc90bf4d448bc0b9b2a8fa0fb02db22eb4d52/pkg/provisioner/ironic/ironic.go#L170), and it could be very likely adopted here

Reply:
So maybe we can have a function in clients for the gofish and we can use the same approach I think (I will try to do some refactor)

Reply:
It's a good idea, but I'm not sure where exactly the client would be cached. In case of ironic clients, we have a provisioner object, but I suspect we don't want to tie the provisioner in here? We can, of course, just accept an external client.

Reply:
I will look at using the Ironic Client

Reply:
After a closer look, I don't think we can use the `clientIronic` it has specific information about ironic, like the endpoint, credentials. I'm looking at adding an http.Client to the  `pkg/bmc/access.go`

Reply:
Agree. As previously discussed, the focus should be on minimizing the number of (new) connections, considering that the `Validate` call will invoked @ every reconcile loop (and there could be many parallel reconciliations based on the value of BMO_CONCURRENCY, by default 3)

Reply:
I think we need to only call it when registering. We should avoid spamming the BMC with Redfish requests, some of them don't like it.

Reply:
@dtantsur @andfasano I've found an approach to add clients directly in `pkg/bmc/<virtualmedia>.go` 
I've highlighted the changes in the code below, let me know if you think this would work, to me it makes sense (it would create the client when `new...AccessDetails`  is called, we can access `client` in the `Validate` function and pass to `ValidateVirtualMedia`.
```
func newRedfishiDracVirtualMediaAccessDetails(parsedURL *url.URL, disableCertificateVerification bool) (AccessDetails, error) {
+	tlsInfo := transport.TLSInfo{
+		TrustedCAFile:      "",
+		InsecureSkipVerify: disableCertificateVerification,
+	}
+	tlsTransport, err := transport.NewTransport(tlsInfo, time.Second*30)

+	var client = http.Client{
+		Transport: tlsTransport,
+	}
	return &redfishiDracVirtualMediaAccessDetails{
		bmcType:                        parsedURL.Scheme,
		host:                           parsedURL.Host,
		path:                           parsedURL.Path,
+		client:                         client,
		disableCertificateVerification: disableCertificateVerification,
+	}, err
}

type redfishiDracVirtualMediaAccessDetails struct {
	bmcType                        string
	host                           string
	path                           string
+      client                         http.Client
	disableCertificateVerification bool
}


func (a *redfishiDracVirtualMediaAccessDetails) Validate(bmcCreds Credentials) error {
	address := getRedfishAddress(a.bmcType, a.host)
+	return ValidateVirtualMedia(address, bmcCreds.Username, bmcCreds.Password,
		a.disableCertificateVerification, a.path, &a.client)
}
```